src/lib/protocols/quic.c

```diff
@@ -1046,14 +1046,11 @@ static int __reassemble(struct ndpi_flow_struct *flow, const u_int8_t *frag,
       return -1; /* Memory error */
     flow->l4.udp.quic_reasm_buf_len = 0;
   }
-  if(flow->l4.udp.quic_reasm_buf_len != frag_offset)
-    return -2; /* Out-of-order, retransmission, overlapping */
   if(frag_offset + frag_len > max_quic_reasm_buffer_len)
     return -3; /* Buffer too small */
 
-  memcpy(&flow->l4.udp.quic_reasm_buf[flow->l4.udp.quic_reasm_buf_len],
-	 frag, frag_len);
-  flow->l4.udp.quic_reasm_buf_len += frag_len;
+  memcpy(&flow->l4.udp.quic_reasm_buf[frag_offset], frag, frag_len);
+  flow->l4.udp.quic_reasm_buf_len = (frag_offset + frag_len > flow->l4.udp.quic_reasm_buf_len) ? frag_offset + frag_len : flow->l4.udp.quic_reasm_buf_len;
```

Comment:
I do not fully understand what your intention was and how that should fix out-of-order packets.

Reply:
Previously, there was an if statement checking if we were getting the frames in order. 

```
if(flow->l4.udp.quic_reasm_buf_len != frag_offset)
    return -2; /* Out-of-order, retransmission, overlapping */
```

I removed that statement and started saving the frames in the buffer according to its offset.
I also modified (https://github.com/ntop/nDPI/pull/1489/commits/f273c5d4476d8abc5437e82374f71dc6c62d0de0) the "is_ch_complete" function. Now we need to check the buffer length and the last written position: if they don't match, frames are missing somewhere in the buffer.

Reply:
LGTM. Waiting for @IvanNardi's approval. =)